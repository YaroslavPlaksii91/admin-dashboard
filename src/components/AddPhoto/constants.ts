export const IMAGE_SIZE = Object.freeze({
  SMALL: 44,
  BIG: 60,
});
