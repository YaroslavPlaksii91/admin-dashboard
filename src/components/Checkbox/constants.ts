export const CHECKBOX_TEST_ID = 'checkbox';
