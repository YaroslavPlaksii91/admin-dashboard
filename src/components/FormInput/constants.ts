export const TOGGLE_BUTTON_TEST_ID = 'toggleButton';
