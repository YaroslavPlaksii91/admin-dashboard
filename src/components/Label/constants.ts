export const LABEL_TYPES = Object.freeze({
  ROUND: 'round',
  SQUARE: 'square',
});

export const LABEL_COLORS = Object.freeze({
  DEFAULT: 'default',
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error',
});
