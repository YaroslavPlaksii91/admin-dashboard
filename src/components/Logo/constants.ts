export const LOGO_TYPES = Object.freeze({
  VERTICAL: 'vertical',
  HORIZONTAL: 'horizontal',
});
