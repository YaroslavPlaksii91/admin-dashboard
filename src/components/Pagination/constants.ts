export const ROWS_PER_PAGE = 8;
export const ROWS_PER_PAGE_OPTIONS = [8, 12, 16, 20];
