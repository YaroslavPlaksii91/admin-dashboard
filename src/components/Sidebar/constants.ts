export const USERS_PAGES = Object.freeze([
  Object.freeze({
    name: 'Overview',
    icon: '/src/assets/icons/sprite.svg#icon-overview',
    link: '/overview',
  }),
  Object.freeze({
    name: 'Tickets',
    icon: '/src/assets/icons/sprite.svg#icon-tickets',
    link: '/tickets',
  }),
  Object.freeze({
    name: 'Ideas',
    icon: '/src/assets/icons/sprite.svg#icon-ideas',
    link: '/ideas',
  }),
  Object.freeze({
    name: 'Contacts',
    icon: '/src/assets/icons/sprite.svg#icon-contacts',
    link: '/contacts',
  }),
  Object.freeze({
    name: 'Agents',
    icon: '/src/assets/icons/sprite.svg#icon-agents',
    link: '/agents',
  }),
  Object.freeze({
    name: 'Articles',
    icon: '/src/assets/icons/sprite.svg#icon-articles',
    link: '/articles',
  }),
]);

export const USERS_SETTINGS = Object.freeze([
  Object.freeze({
    name: 'Settings',
    icon: '/src/assets/icons/sprite.svg#icon-settings',
    link: '/settings',
  }),
  Object.freeze({
    name: 'Subscription',
    icon: '/src/assets/icons/sprite.svg#icon-subscription',
    link: '/subscription',
  }),
]);
