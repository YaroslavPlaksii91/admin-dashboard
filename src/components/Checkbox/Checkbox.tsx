import { FC, useState } from 'react';
import { Checkbox, SvgIcon, Box } from '@mui/material';

import { CHECKBOX_TEST_ID } from './constants';

export const CheckboxComponent: FC = () => {
  const [checked, setChecked] = useState(false);

  const handleChange = () => {
    setChecked(!checked);
  };

  return (
    <label>
      <Checkbox
        checked={checked}
        onChange={handleChange}
        sx={{ display: 'none' }}
        data-testid={CHECKBOX_TEST_ID}
      />
      <Box
        sx={{
          height: '20px',
          cursor: 'pointer',
        }}
      >
        <SvgIcon
          titleAccess={checked ? 'Task completed' : 'Task not completed'}
          sx={{ width: '20px', height: '20px' }}
        >
          <use
            href={`/src/assets/icons/sprite.svg#icon-${
              checked ? 'active' : 'inactive'
            }Checkbox`}
          ></use>
        </SvgIcon>
      </Box>
    </label>
  );
};
