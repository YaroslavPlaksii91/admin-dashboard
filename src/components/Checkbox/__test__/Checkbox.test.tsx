import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import { CheckboxComponent } from '../Checkbox';
import { CHECKBOX_TEST_ID } from '../constants';

describe('Checkbox component', () => {
  afterEach(cleanup);

  it('should be rendered', () => {
    const { asFragment } = render(<CheckboxComponent />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(<CheckboxComponent />);

    const checkboxElement = screen.getByTestId(CHECKBOX_TEST_ID);

    expect(checkboxElement).toBeInTheDocument();
  });

  it('toggles the checkbox when clicked', () => {
    render(<CheckboxComponent />);

    const checkboxElement = screen.getByTestId(CHECKBOX_TEST_ID);

    expect(checkboxElement).not.toHaveClass('Mui-checked');

    fireEvent.click(checkboxElement);

    expect(checkboxElement).toHaveClass('Mui-checked');
  });
});
