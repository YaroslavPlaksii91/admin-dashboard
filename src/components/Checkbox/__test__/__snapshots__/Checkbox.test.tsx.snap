// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should be rendered 1`] = `
<DocumentFragment>
  <label>
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-le7zt4-MuiButtonBase-root-MuiCheckbox-root"
      data-testid="checkbox"
    >
      <input
        class="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
    <div
      class="MuiBox-root css-yhnhoj"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xhbeow-MuiSvgIcon-root"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
      >
        <use
          href="/src/assets/icons/sprite.svg#icon-inactiveCheckbox"
        />
        <title>
          Task not completed
        </title>
      </svg>
    </div>
  </label>
</DocumentFragment>
`;

exports[`Label component should be rendered 1`] = `
<DocumentFragment>
  <label>
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-le7zt4-MuiButtonBase-root-MuiCheckbox-root"
      data-testid="checkbox"
    >
      <input
        class="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
    <div
      class="MuiBox-root css-yhnhoj"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xhbeow-MuiSvgIcon-root"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
      >
        <use
          href="/src/assets/icons/sprite.svg#icon-inactiveCheckbox"
        />
        <title>
          Task not completed
        </title>
      </svg>
    </div>
  </label>
</DocumentFragment>
`;
